pkg/scheduler/objects/node.go

```diff
@@ -377,6 +382,8 @@ func (sn *Node) ReplaceAllocation(allocationKey string, replace *Allocation, del
 	// The allocatedResource and availableResource should be updated in the same way
 	sn.allocatedResource.AddTo(delta)
 	sn.availableResource.SubFrom(delta)
+	sn.allocatedResource.Prune()
+	sn.availableResource.Prune()
```

Comment:
The alloc should not have zeros in it. The only prune that is needed is when we do the `SubFrom`

Reply:
Addressed in latest PR, thanks!